tools/cryptogenotyper/cryptogenotyper.xml

```diff
@@ -0,0 +1,141 @@
+        #set $ref_file = $reference.db.name
+     #end if
+
+     #if $primers["seqtype"] == "contig"
+        ln -s "${$primers.abi_input["forward"]}" "${primers.abi_input.name}_forward.ab1" &&
+        ln -s "${$primers.abi_input["reverse"]}" "${primers.abi_input.name}_reverse.ab1" &&
+        cryptogenotyper -i "." -m "$marker" -t "$primers.seqtype" -f "forward" -r "reverse"
+        #if $reference.ref == "no"
+          --databasefile $ref_file
+        #end if
+     #else
+        ln -s "${primers.abi_input}" "${primers.abi_input.element_identifier}" &&
+        cryptogenotyper -i "./${primers.abi_input.element_identifier}" -m "$marker" -t "$primers.seqtype" -f "" -r ""
+        #if $reference.ref == "no"
+          --databasefile $ref_file
+        #end if
+     #end if
+     #if $outputheader == "no"
+           --noheaderline
+     #end if
+     -o "result";
+     ]]>
+  </command>
+  <inputs>
+      <param name="marker" type="select" label="Marker">
+          <option value="18S">SSU rRNA</option>
+          <option value="gp60">gp60</option>
+      </param>
+      <conditional name="reference">
+        <param name="ref" type="select" label="Use default reference file?">
+            <option value="yes">Yes</option>
+            <option value="no">No</option>
+        </param>
+        <when value="yes">
+        </when>
+        <when value="no">
+            <param name="db" type="data" format="fasta" label="Reference Database File:" help=".fa fasta file type"/>
+        </when>
+      </conditional>
+      <conditional name="primers">
+        <param name="seqtype" type="select" label="Type of Sequences">
+            <option value="forward">Forward Only</option>
+            <option value="reverse">Reverse Only</option>
+            <option selected="true" value="contig">Contig</option>
+        </param>
+        <when value="contig">
+          <param name="abi_input" type="data_collection" collection_type="paired" format="ab1" label="Paired Sequencing File(s)" help=".ab1 file type"/>
+        </when>
+        <when value="forward">
+          <param name="abi_input" type="data" format="ab1" label="Forward Sequencing File(s)" help=".ab1 file type"/>
+        </when>
+        <when value="reverse">
+          <param name="abi_input" type="data" format="ab1" label="Reverse Sequencing File(s)" help=".ab1 file type"/>
+        </when>
+     </conditional>
+     <param name="outputheader" type="select" value="yes" label="Output header line in the report?">
+        <option value="yes">Yes</option>
+        <option value="no">No</option>
+     </param>
+
+
+  </inputs>
+  <outputs>
+    <data name="outfile" format="fasta" from_work_dir="result_cryptogenotyper_report.fa" label="${tool.name}:${on_string}:fastas"/>
+    <data name="outfile_report" format="tabular" from_work_dir="result_cryptogenotyper_report.txt" label="${tool.name}:${on_string}:reports"/>
+  </outputs>
+
+  <tests>
+    <test>
```

Comment:
can you please add `expected_num_output="2"`

Reply:
Implemented. Great security feature to add. Did not know about it.